tests/ignite/metrics/test_fbeta.py

```diff
@@ -71,19 +94,22 @@ def update_fn(engine, batch):
         y_true_batch = next(y_true_batch_values)
         y_pred_batch = next(y_pred_batch_values)
         if output_transform is not None:
-            return {"y_pred": torch.from_numpy(y_pred_batch), "y": torch.from_numpy(y_true_batch)}
-        return torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+            return {"y_pred": y_pred_batch, "y": y_true_batch}
+        return y_pred_batch, y_true_batch
 
     evaluator = Engine(update_fn)
 
-    f2 = Fbeta(beta=2.0, average=average, precision=p, recall=r, output_transform=output_transform)
+    device = None if p is not None and r is not None else available_device
+    f2 = Fbeta(beta=2.0, average=average, precision=p, recall=r, output_transform=output_transform, device=device)
+
     f2.attach(evaluator, "f2")
 
     data = list(range(n_iters))
     state = evaluator.run(data, max_epochs=1)
 
-    f2_true = fbeta_score(y_true, np.argmax(y_pred, axis=-1), average="macro" if average else None, beta=2.0)
-    np.testing.assert_allclose(np.array(f2_true), np.array(state.metrics["f2"]))
+    f2_true = fbeta_score(y_true, torch.argmax(y_pred, dim=-1), average="macro" if average else None, beta=2.0)
+    f2_true = np.float32(f2_true) if available_device == "mps" else f2_true
```

Comment:
Let's explicitly convert y_true and torch.argmax(y_pred, dim=-1) to numpy passing by cpu.
And remove `np.float32(f2_true)`:
```suggestion
    f2_true = fbeta_score(y_true, torch.argmax(y_pred, dim=-1), average="macro" if average else None, beta=2.0)
```

Reply:
something like that?

```python
    y_true_np = y_true.detach().cpu().numpy()
    y_pred_np = torch.argmax(y_pred, dim=-1).detach().cpu().numpy()
    f2_true = fbeta_score(y_true_np, y_pred_np, average="macro" if average else None, beta=2.0)
```